### PR TITLE
Add option to disable fake quant for 8da4w QAT

### PR DESCRIPTION
**Pull request by andrewor14**
Summary: This feature helps with model convergence during QAT. The user can disable observation/fake quant for the first N steps and renable them later, allowing the activation and weight values to stabilize before applying quantization.

Test Plan:
python test/quantization/test_qat.py -k test_qat_8da4w_quantizer_disable_fake_quant
python test/quantization/test_qat.py -k test_qat_8da4w_quantizer_disable_fake_quant_backward

Reviewers: jerryzh168, cpuhrsch

Subscribers: jerryzh168, cpuhrsch, supriyar

### CODE REVIEW

File: test/quantization/test_qat.py [hunk 1]
```diff
@@ -200,6 +200,101 @@ def test_qat_8da4w_quantizer(self):
...
+
+        # Renable fake quant
+        qat_model.apply(enable_8da4w_fake_quant)
+        self.assertTrue(qat_model.linear1._fake_quant_enabled)
+        self.assertTrue(qat_model.linear2._fake_quant_enabled)
+        self.assertTrue(qat_model.sub.linear._fake_quant_enabled)
+
+        # Fake quant should be applied as normal
+        quantizer2 = Int8DynActInt4WeightQATQuantizer(groupsize=group_size)
+        qat_model2 = quantizer2.prepare(m)
```
Comment (jerryzh168) [inline test/quantization/test_qat.py:241]: nit: probably need to copy m in the beginning since `prepare` may change `m`

**Review by jerryzh168 — APPROVED** [review]

**Comment by andrewor14** [comment]
Actually I had to remove `enable_observer` because saving the qparams increased the memory usage significantly and caused OOM in my experiments. I think this is not really needed for 8da4w anyway because there's no state in the observer

**Comment by jerryzh168** [comment]
@andrewor14 maybe you can unlink the diff

**Comment by andrewor14** [comment]
> @andrewor14 maybe you can unlink the diff

How can I do that?

**Pull request merged**